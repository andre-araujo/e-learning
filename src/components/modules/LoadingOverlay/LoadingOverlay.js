import React from 'react';
import { bool } from 'prop-types';

import Loading from '../../elements/Loading';

import {
  Container,
} from './LoadingOverlay.styles';

function LoadingOverlay({
  loading,
}) {
  return (
    <Container active={loading}>
      <Loading />
    </Container>
  );
}

LoadingOverlay.defaultProps = {
  loading: false,
};

LoadingOverlay.propTypes = {
  loading: bool,
};

export default LoadingOverlay;
