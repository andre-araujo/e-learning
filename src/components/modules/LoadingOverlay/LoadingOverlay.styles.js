import styled, { css } from 'styled-components';

export const Container = styled.div`
  align-items: center;
  background: rgba(255, 255, 255, 0.8);
  bottom: 0;
  display: flex;
  left: 0;
  opacity: 0;
  pointer-events: none;
  position: fixed;
  justify-content: center;
  right: 0;
  top: 0;
  z-index: 100;
  transition: 0.1s opacity;
  ${({ active }) => active && css`
      opacity: 1;
      pointer-events: all;
  `}
`;
