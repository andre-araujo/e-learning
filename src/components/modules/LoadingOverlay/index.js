import { connect } from 'react-redux';

import LoadingOverlay from './LoadingOverlay';

const mapStateToProps = ({ globalLoading }) => ({ loading: globalLoading.loading });

export default connect(mapStateToProps)(LoadingOverlay);
