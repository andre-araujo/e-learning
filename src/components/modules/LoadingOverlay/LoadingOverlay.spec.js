import React from 'react';
import { shallow } from 'enzyme';

import LoadingOverlay from './LoadingOverlay';

describe('LoadingOverlay component', () => {
  it('should match snapshot', () => {
    expect(shallow(<LoadingOverlay />)).toMatchSnapshot();
  });

  it('should match snapshot with loading', () => {
    expect(shallow(<LoadingOverlay loading />)).toMatchSnapshot();
  });
});
