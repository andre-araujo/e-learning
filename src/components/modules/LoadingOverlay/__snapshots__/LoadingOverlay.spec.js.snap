// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay component should match snapshot 1`] = `
<styled.div
  active={false}
>
  <Loading />
</styled.div>
`;

exports[`LoadingOverlay component should match snapshot with loading 1`] = `
<styled.div
  active={true}
>
  <Loading />
</styled.div>
`;
